mainjsdoshop

```diff
@@ -2,18 +2,43 @@
 
 
 var DonutShop = function(locationName, avgMin, avgMax, avgPerCustomer) {
-  this.locationName = locationName
+  this.locationName = locationName;
   this.avgMin = avgMin;
   this.avgMax = avgMax;
-  this.avgPerCustoemr = avgPerCustomer;
-  this.myArray = [];
-  this.total = 0;
-
+  this.avgPerCustomer = avgPerCustomer;
 
 DonutShop.prototype.nutPerHour = function() {
-this.total = this.myArray
-this.myArray = Math.floor(Math.random() * (this.avgMax-this.avgMin + 1) + (this.avgMin * this.avgPerCustomer));
-}
+  this.totalSales = 0;
+  this.hourlyNutArray = [];
+  for(var i = 0; i < 12; i++) {
+  this.hourlyNutArray.push(Math.floor((Math.random() * (this.avgMax-this.avgMin + 1) + this.avgMin) * this.avgPerCustomer));
+  this.totalSales = this.totalSales + this.hourlyNutArray[i];
+  }
+
+return this.hourlyNutArray;
+  };
+
+DonutShop.prototype.render = function () {
+  var getTable = document.getElementById('donuttable');
+  var newRow = document.createElement('tr');
+
+  newRow.id = this.locationName;
+  newRow.innerHTML = this.locationName;
+  getTable.appendChild(newRow);
+  this.nutPerHour();
+
+  for(var i = 0; i < 12; i++) {
+    var newCell = document.createElement('td');
+    newCell.innerHTML = this.hourlyNutArray[i];
+    newRow.appendChild(newCell);
+  };
+
+  var newCell = document.createElement('td');
```

Comment:
This will work, but you're reusing a variable name that's similarly scoped to the for loop in this function. Just be careful that you're not causing scope issues; usually accomplished by naming variables in a thoughtful way.